File: src/main/java/org/noxet/noxetserver/commands/admin/TogglePreserve.java
package org.noxet.noxetserver.commands.admin;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.noxet.noxetserver.NoxetServer;
import org.noxet.noxetserver.messaging.ErrorMessage;
import org.noxet.noxetserver.messaging.SuccessMessage;

public class TogglePreserve implements CommandExecutor {
    @Override
    @SuppressWarnings("NullableProblems")
    public boolean onCommand(CommandSender commandSender, Command command, String s, String[] strings) {
        if(!commandSender.isOp()) {
            new ErrorMessage(ErrorMessage.ErrorType.PERMISSION, "Only operators can toggle preservation mode.").send(commandSender);
            return true;
        }

        NoxetServer.shouldAllowWorldPreservation = !NoxetServer.shouldAllowWorldPreservation;
        new SuccessMessage((NoxetServer.shouldAllowWorldPreservation ? "Enabled" : "Disabled") + " world preservation.").send(commandSender);

        return true;
    }
}

File: src/main/java/org/noxet/noxetserver/commands/admin/TogglePreserve.scala
package org.noxet.noxetserver.commands.admin

import org.bukkit.command.{Command, CommandExecutor, CommandSender}
import org.noxet.noxetserver.NoxetServer
import org.noxet.noxetserver.commands.RegisteredCommand
import org.noxet.noxetserver.messaging.{ErrorMessage, SuccessMessage}

object TogglePreserve extends OperatorCommandExecutor, RegisteredCommand("toggle-preserve", this):
  override def onOperatorCommand(commandSender: CommandSender, command: Command, s: String, strings: Array[String]): Boolean =
    NoxetServer.shouldAllowWorldPreservation ^= true
    SuccessMessage(
      s"${if NoxetServer.shouldAllowWorldPreservation then "Enabled" else "Disabled"} world preservation."
    ).send(commandSender)
    true